### PR TITLE
Implemented Login & Register module using social_auth.django

### LINKED ISSUE #6: Login/Register
Create a Login/Register system to the site, using Google and Facebook authentication.

Some helpful links:

With [`social-auth`](https://github.com/python-social-auth/social-app-django):
https://www.digitalocean.com/community/tutorials/django-authentication-with-facebook-instagram-and-linkedin
https://realpython.com/adding-social-authentication-to-django/
https://medium.com/trabe/oauth-authentication-in-django-with-social-auth-c67a002479c1

With [`all-auth`](https://django-allauth.readthedocs.io/en/latest/installation.html)
https://medium.com/@whizzoe/in-5-mins-set-up-google-login-to-sign-up-users-on-django-e71d5c38f5d5
https://medium.com/@jinkwon711/django-allauth-facebook-login-b536444cbc6b

### PR DESCRIPTION
**Pull request by 11galpid**
Implemented Login & Register module using social_auth.django

Supports both Facebook & Google (Can be extended further) using the host:port "localhost:8000" (That's what both service providers currently allow, if and when we'll have a domain this can be extended further, but for the development I think that should suffice)

Currently saves: Full name, Email and Picture

Also note, the HTML just a first draft, to give the general feeling of the login/register, it is just a "mockup" and will be replaced when we decide on a template for the website

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,4 +68,17 @@ Guy Itzhaki, Tarel Madar, Gal Lapid, Alon Weissfeld, Omri Rosner
 1. Install VirtualBox and Vagrant.
 2. Clone this repository.
 3. Open a shell prompt, and switch to the location of the repository.
-4. Run 'vagrant up'. Once it finished, you're all set!
```
Comment (ifireball) [inline README.md]: The following does not seem related to login/registration - does this belong in a different PR?

File: sellotape/sellotape_dj/sellotape_dj/urls.py [hunk 2]
```diff
@@ -14,8 +14,9 @@
     2. Add a URL to urlpatterns:  path('blog/', include('blog.urls'))
 """
 from django.contrib import admin
-from django.urls import path
+from django.urls import path, include
 
 urlpatterns = [
     path('admin/', admin.site.urls),
+    path('auth_app/', include('sellotape_dj.auth_app.urls')),
```
Comment (ifireball) [inline sellotape/sellotape_dj/sellotape_dj/urls.py]: I wonder about auth being its own app - shouldn't this be embedded into the actual app that needs authentication?

File: sellotape/sellotape_dj/sellotape_dj/auth_app/templates/home.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<!-- home.html -->
+
+    {% extends 'base.html' %}
+    {% block title %} Home {% endblock %}
+    {% block content %}
+      <div class="row">
+
+            <!-- Add from here -->
+        {% for ass in backends.associated %}
```
Comment (ifireball) [inline sellotape/sellotape_dj/sellotape_dj/auth_app/templates/home.html]: I'm seeing what seems to be code repetition below, could you avoid that? 

Also `ass` is not a good variable name - it does not help me understand what it contains.

File: sellotape/sellotape_dj/sellotape_dj/settings.py [hunk 4]
```diff
@@ -118,3 +128,38 @@
...
+    'social_core.pipeline.social_auth.social_user',
+    'social_core.pipeline.user.get_username',
+    'social_core.pipeline.social_auth.associate_by_email',
+    'social_core.pipeline.user.create_user',
+    'social_core.pipeline.social_auth.associate_user',
+    'social_core.pipeline.social_auth.load_extra_data',
+    'social_core.pipeline.user.user_details',
+)
+
+SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = '87810693431-99al9o78efneb8dpuvhkkldmdon2n7tm.apps.googleusercontent.com'
```
Comment (ifireball) [inline sellotape/sellotape_dj/sellotape_dj/settings.py]: are you sure you want to leak your key to the entire internet?

File: sellotape/sellotape_dj/sellotape_dj/settings.py [hunk 5]
```diff
@@ -118,3 +128,38 @@
...
+  'fields': 'id, name, email, picture.type(large)'
+}
+SOCIAL_AUTH_GOOGLE_OAUTH2_EXTRA_DATA = [
+    ('name', 'name'),
+    ('email', 'email'),
+    ('picture', 'picture')
+]
+
+SOCIAL_AUTH_FACEBOOK_KEY = "337762567421238"        # App ID
+SOCIAL_AUTH_FACEBOOK_SECRET = "f90f36fc0409c0c350a1767e873716d7"  # App Secret
```
Comment (ifireball) [inline sellotape/sellotape_dj/sellotape_dj/settings.py]: same for the secret

File: README.md (hunk 1 above)
Comment (11galpid) [reply README.md]: Yes, you're right.
I felt it was redundant to open a new PR for such a small change, and this is related to Authentication & Login, but I will do so if necessary.

File: sellotape/sellotape_dj/sellotape_dj/urls.py (hunk 2 above)
Comment (11galpid) [reply sellotape/sellotape_dj/sellotape_dj/urls.py]: It's only for this stage of development, when we have enough code in our project this will change.

File: sellotape/sellotape_dj/sellotape_dj/auth_app/templates/home.html (hunk 3 above)
Comment (11galpid) [reply sellotape/sellotape_dj/sellotape_dj/auth_app/templates/home.html]: Yes, you're right. In the development stages the variables were different for different providers, but it eventually changed and I didn't notice this. Updating in next commit.

File: sellotape/sellotape_dj/sellotape_dj/settings.py (hunk 4 above)
Comment (11galpid) [reply sellotape/sellotape_dj/sellotape_dj/settings.py]: I didn't think it will matter as we're in development stages, and this is only for localhost. But of course, as a best practice it should be avoided. I'll remove this and send them in the Slack group chat for better security.

**Review by ifireball — CHANGES_REQUESTED** [review]
I'm having a hard time understanding how the app would act without the secret keys - you should probably document that somewhere, and also what developers need to do to make the keys available in their environment.

I recommend not introducing this for the next milestone - as it stands, the dev environment setup does not seem to be as automatic as required by the milestone.
